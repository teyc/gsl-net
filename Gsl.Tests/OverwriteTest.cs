using System.IO;
using Xunit;
using Xunit.Abstractions;

namespace Gsl.Tests
{
    public class OverwriteTest : TestBase
    {
        public OverwriteTest(ITestOutputHelper log) : base(log)
        {
        }

        [Fact]
        public void DoNotOverwriteFilesIfSpecialTextIsPresent()
        {
            const string template = "data/overwrite.gsl";
            const string data = "data/overwrite.json";

            // Given that an output file already exists, we don't want to overwrite it
            _fileSystem.File.WriteAllText("Overwrite.cs",
                File.ReadAllText(Path.DataFile("data/OverwriteTest.DoNotOverwriteFilesIfSpecialTextIsPresent.approved.txt")));

            TemplateWithSingleFileOutput(template, data);
        }
    }
}
