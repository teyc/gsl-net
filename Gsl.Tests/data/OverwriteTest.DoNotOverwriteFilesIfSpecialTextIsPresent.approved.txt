// DO NOT OVERWRITE
// Overwrite test
// If the target output file contains a particular string, then write 
// the output to and alternate location
public class Overwrite
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Age { get; set; }
    public string JobTitle { get; set; }
}